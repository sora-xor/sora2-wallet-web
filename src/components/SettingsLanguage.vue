<template>
  <wallet-base :title="t('settings.menu.Language.title')" show-back @back="handleBack">
    <div class="wallet-settings-language">
      <div
        v-for="(locale, index) in sortedLocales"
        :key="index"
        @click="handleChangeLocale(locale)"
      >
        <div class="wallet-settings-language-item s-flex">
          <div class="wallet-settings-language-item-text s-flex">
            <span
              :class="[
                'wallet-settings-language-item-text_main',
                currentLocale === locale ? 'active' : ''
              ]"
            >
              {{ avaliableLocales[locale] }}
            </span>
            <span
              v-if="currentLocale !== locale"
              class="wallet-settings-language-item-text_secondary"
            >
              {{ t(`settings.language.${locale}`) }}
            </span>
          </div>
          <s-icon
            v-if="currentLocale === locale"
            class="wallet-settings-language-item_icon active"
            name="check-mark"
            :size="16"
          />
        </div>
        <s-divider v-if="index !== sortedLocales.length - 1" class="wallet-settings-language-item_divider" />
      </div>
    </div>
  </wallet-base>
</template>

<script lang="ts">
import { Component, Mixins, Prop } from 'vue-property-decorator'
import { Action, State } from 'vuex-class'
import sortBy from 'lodash/fp/sortBy'

import TranslationMixin from './mixins/TranslationMixin'
import WalletBase from './WalletBase.vue'
import { Languages, RouteNames } from '../consts'

@Component({
  components: {
    WalletBase
  }
})
export default class SettingsLanguage extends Mixins(TranslationMixin) {
  @State(state => state.Settings.locales) locales!: Array<string>
  @State(state => state.Settings.locale) currentLocale!: string

  @Action navigate
  @Action setLocale

  get sortedLocales () {
    // Move currentLocale to the top
    return sortBy((lang) => lang === this.currentLocale ? 0 : 1, this.locales)
  }

  get avaliableLocales () {
    const locales = {}
    Object.values(Languages).forEach(lang => {
      locales[lang] = this.t(`settings.language.${lang}`)
    })
    return locales
  }

  handleChangeLocale (locale: string): void {
    if (locale === this.currentLocale) {
      return
    }
    this.setLocale({ locale })
    this.$root.$i18n.locale = locale
  }

  handleBack (): void {
    this.navigate({ name: RouteNames.WalletSettings })
  }
}
</script>

<style scoped lang="scss">
.wallet-settings-language {
  &-item {
    justify-content: space-between;
    padding: $basic-spacing_small 0;
    &-text {
      flex-direction: column;
      &_main {
        font-size: $font-size_normal;
        line-height: $line-height_medium;
        &.active {
          color: var(--s-color-button-tertiary-color);
        }
      }
      &_secondary {
        font-size: $font-size_small;
        color: var(--s-color-base-content-tertiary);
        line-height: $line-height_medium;
      }
    }
    &_icon {
      margin: auto 0;
      &.active {
        color: var(--s-color-button-tertiary-color);
      }
    }
    &_divider {
      margin: unset;
    }
    &:hover {
      cursor: pointer;
      .wallet-settings-language-item-text_main {
        color: var(--s-color-button-tertiary-color);
      }
    }
  }
}
</style>
