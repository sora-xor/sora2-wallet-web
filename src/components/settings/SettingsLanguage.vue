<template>
  <wallet-base :title="t('settings.menu.Language.title')" show-back @back="handleBack">
    <div class="wallet-settings-language">
      <div
        v-for="(locale, index) in locales"
        :key="index"
        @click="setLocale({ locale })"
      >
        <div class="wallet-settings-language-item s-flex">
          <div class="wallet-settings-language-item-text s-flex">
            <span
              :class="[
                'wallet-settings-language-item-text_main',
                currentLocale === locale ? 'active' : ''
              ]"
            >{{ avaliableLocales[locale] }}</span>
            <span
              v-if="currentLocale !== locale"
              class="wallet-settings-language-item-text_secondary"
            > {{ t(`settings.language.${locale}`) }}</span>
          </div>
          <s-icon
            v-if="currentLocale === locale"
            class="wallet-settings-language-item_icon active"
            name="check-mark"
            :size="16"
          />
        </div>
        <s-divider v-if="index !== locales.length - 1" style="margin: unset" />
      </div>
    </div>
  </wallet-base>
</template>

<script lang="ts">
import { Component, Mixins, Prop } from 'vue-property-decorator'
import { Action, State } from 'vuex-class'

import TranslationMixin from '../mixins/TranslationMixin'
import WalletBase from '../WalletBase.vue'
import { Languages, RouteNames } from '../../consts'

@Component({
  components: {
    WalletBase
  }
})
export default class SettingsLanguage extends Mixins(TranslationMixin) {
  @State(state => state.Settings.locales) locales
  @State(state => state.Settings.locale) currentLocale

  @Action navigate
  @Action setLocale

  get avaliableLocales () {
    return {
      [Languages.EN]: 'English'
    }
  }

  handleBack (): void {
    this.navigate({ name: RouteNames.WalletSettings })
  }
}
</script>

<style scoped lang="scss">
@import '../../styles/typography';
@import '../../styles/layout';
@import '../../styles/soramitsu-variables';

.wallet-settings-language {
  &-item {
    justify-content: space-between;
    padding: $basic-spacing_small 0;
    &-text {
      flex-direction: column;

      &_main {
        font-size: $font-size_normal;
        line-height: 1.8;
        &.active {
          color: $s-color-button-tertiary-color;
        }
      }
      &_secondary {
        font-size: $font-size_small;
        color: $s-color-base-content-tertiary;
        line-height: 1.8;
      }
    }

    &_icon {
      margin: auto 0;

      &.active {
        color: $s-color-button-tertiary-color;
      }
    }

    &:hover {
      cursor: pointer;
      .wallet-settings-language-item-text_main {
        color: $s-color-button-tertiary-color;
      }
    }
  }
}
</style>
