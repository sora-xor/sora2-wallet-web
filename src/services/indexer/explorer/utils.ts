import { formatStringNumber } from '../../../util';

import type { SubqueryAssetEntity } from '../subquery/types';
import type { SubsquidAssetEntity } from '../subsquid/types';
import type { FiatPriceObject, UpdatesStream, PoolApyObject } from '../types';

export function parseAssetFiatPrice(entity: SubsquidAssetEntity | SubqueryAssetEntity): FiatPriceObject {
  const acc = {};
  const id = entity.id;
  const priceFPNumber = formatStringNumber(entity.priceUSD);
  const isPriceFinity = priceFPNumber.isFinity();
  if (isPriceFinity) {
    acc[id] = priceFPNumber.toCodecString();
  }
  return acc;
}

export function parsePriceStreamUpdate(entity: UpdatesStream): Nullable<FiatPriceObject> {
  if (!entity?.data) return null;

  const data = JSON.parse(entity.data);

  return Object.entries(data).reduce((acc, [id, price]) => {
    const priceFPNumber = formatStringNumber(price as string);
    const isPriceFinity = priceFPNumber.isFinity();
    if (isPriceFinity) {
      acc[id] = priceFPNumber.toCodecString();
    }
    return acc;
  }, {});
}

export function parseApyStreamUpdate(entity: UpdatesStream): PoolApyObject {
  const data = entity?.data ? JSON.parse(entity.data) : {};

  return Object.entries(data).reduce((acc, [id, apy]) => {
    const strategicBonusApyFPNumber = formatStringNumber(apy as string);
    const isStrategicBonusApyFinity = strategicBonusApyFPNumber.isFinity();
    if (isStrategicBonusApyFinity) {
      acc[id] = strategicBonusApyFPNumber.toCodecString();
    }
    return acc;
  }, {});
}
