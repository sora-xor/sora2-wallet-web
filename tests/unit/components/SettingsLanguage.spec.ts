import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'

import SettingsLanguage from '@/components/SettingsLanguage.vue'
import { TranslationMock, SoramitsuElementsImport } from '../../utils'

const localVue = createLocalVue()
localVue.use(Vuex)
const store = new Vuex.Store({
  modules: {
    Router: {
      actions: {
        navigate: jest.fn()
      }
    },
    Settings: {
      state: {
        locales: ['en', 'ru'],
        locale: 'en'
      },
      actions: {
        setLocale: jest.fn()
      }
    }
  } as any
})

describe('SettingsLanguage.vue', () => {
  beforeEach(() => {
    SoramitsuElementsImport(localVue)
    TranslationMock(SettingsLanguage)
  })

  it('should renders correctly', () => {
    const wrapper = shallowMount(SettingsLanguage, { localVue, store })
    expect(wrapper.element).toMatchSnapshot()
  })
})
