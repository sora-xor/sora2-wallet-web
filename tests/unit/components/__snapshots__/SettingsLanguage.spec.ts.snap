// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsLanguage.vue should renders correctly 1`] = `
<wallet-base-stub
  showback="true"
  title=""
>
  <div
    class="wallet-settings-language"
  >
    <div>
      <div
        class="wallet-settings-language-item s-flex"
      >
        <div
          class="wallet-settings-language-item-text s-flex"
        >
          <span
            class="wallet-settings-language-item-text_main active"
          >
            
            
          
          </span>
           
          <!---->
        </div>
         
        <s-icon-stub
          class="wallet-settings-language-item_icon active"
          name="check-mark"
          size="16"
        />
      </div>
       
      <s-divider-stub
        class="wallet-settings-language-item_divider"
        contentposition="center"
        direction="horizontal"
        type="secondary"
      />
    </div>
    <div>
      <div
        class="wallet-settings-language-item s-flex"
      >
        <div
          class="wallet-settings-language-item-text s-flex"
        >
          <span
            class="wallet-settings-language-item-text_main"
          >
            
            
          
          </span>
           
          <span
            class="wallet-settings-language-item-text_secondary"
          >
            
            
          
          </span>
        </div>
         
        <!---->
      </div>
       
      <!---->
    </div>
  </div>
</wallet-base-stub>
`;
